app/src/main/java/ckmah/mangoreader/UserLibraryHelper.java

```diff
@@ -1,162 +1,113 @@
-            SQLiteDatabase userLibraryDb = helper.getWritableDatabase();
-            // helper.onUpgrade(userLibraryDb, userLibraryDb.getVersion(), 1000); // DEBUG PURPOSES ONLY
-            DaoMaster daoMaster = new DaoMaster(userLibraryDb);
-            DaoSession daoSession = daoMaster.newSession();
-            userLibraryMangaDao = daoSession.getUserLibraryMangaDao();
-        }
-        return userLibraryMangaDao;
-    }
 
+    public static final String USER_LIBRARY_DB = "user-library";
     private static String added;
     private static String removed;
 
-    public static boolean isInLibrary(Context context, MangaEdenMangaListItem item) {
-        // If any manga ids are matched, then manga must be in library.
-        QueryBuilder qb = getDao(context).queryBuilder();
-        qb.where(UserLibraryMangaDao.Properties.MangaEdenId.eq(item.id));
-        return qb.list().size() > 0;
+    public static List<Manga> findAllFavoritedManga() {
+        List<Manga> response = new ArrayList<>();
+        for (String key : Paper.book(USER_LIBRARY_DB).getAllKeys()) {
+            Manga m = Paper.book(USER_LIBRARY_DB).read(key);
+            if (m.favorite) {
+                response.add(m);
+            }
+            //TODO: else check if flagged for deletion
+        }
+        return response;
     }
 
-    /**
-     * Creates dialog for user to select library category to add to.
-     *
-     * @param m
-     */
-    public static boolean addToLibrary(final MangaEdenMangaListItem m, final View button, final Activity activity, final CardLayoutAdapter adapter, final int position) {
-        String genres = TextUtils.join("\t", m.genres);
+    public static void addToLibrary(final Manga m, final View button, final Activity activity, boolean showUndo, final CardLayoutAdapter adapter, final int position) {
+
         added = "\"" + m.title + "\" added to your library under \"Plan to Read\"";
         removed = "\"" + m.title + "\" removed from your library.";
-        final UserLibraryManga mangaItem = new UserLibraryManga(
-                m.id,
-                activity.getResources().getStringArray(R.array.library_categories)[0],
-                m.title,
-                m.imageUrl,
-                genres,
-                m.status,
-                m.lastChapterDate,
-                m.hits);
-
-        try { // insert and show snackbar with undo, return true if successful, false otherwise
-            getDao(activity).insert(mangaItem);
-            Snackbar
-                .make(findMyView(activity), added, Snackbar.LENGTH_LONG)
-                .setAction("UNDO", new View.OnClickListener() {
+
+        m.favorite = true;
+        Paper.book(USER_LIBRARY_DB).write(m.id, m);
+        button.setSelected(true);
+
+        MangaEden.getMangaEdenService(activity)
+                .getMangaDetails(m.id)
+                .enqueue(new Callback<MangaEdenMangaDetailItem>() {
```

Comment:
Makes a request to download all the manga metadata. This may be a performance/bandwidth issue, especially if someone repeatedly toggles the bookmark icon. Maybe should check a last updated timestamp, and only dl when > some elapsed time, otherwise skip the request?